src/js/hooks/useTableCheckboxes.js

```diff
@@ -47,6 +49,13 @@ const useTableCheckboxes = () => {
     headerCheckboxState.value,
   ]);
 
+  useEffect(() => {
+    if (checkedCheckboxes.length > 0) {
+      setCheckedCheckboxes([]);
+      setHeaderCheckboxState({ indeterminate: false, value: false });
+    }
+  }, [currentLocation?.id]);
+
```

Comment:
Add a comment about the purpose of this useEffect. If you want you can also move these two lines to a function called `resetCheckboxes` for better readability